### PR TITLE
Khara

### PR DESCRIPTION
**Pull request by KharaHL**

### CODE REVIEW

**Review by fresh-ash — APPROVED** [review]

File: src/main/java/page/mentors/EditMentorsDetailsPage.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    @FindBy (xpath =ADD_COURSE_BUTTON_XPATH)
+    WebElement addCourseButton;
+    @FindBy (xpath =DISABLE_BUTTON_XPATH)
+    WebElement disableButton;
+    @FindBy (xpath =CLEAR_BUTTON_XPATH)
+    WebElement clearButton;
+    @FindBy (xpath =SAVE_BUTTON_XPATH)
+    WebElement saveButton;
+
+    public EditMentorsDetailsPage viewMentorsDetailTab(){
```
Comment (OlekLiakh) [inline src/main/java/page/mentors/EditMentorsDetailsPage.java]: As result of execution of this metod you'll be have MentorDetialsPage. So return type will be new MentorDetailsPage
Maybe change name of method to openMentorDetailPage or activateMentorDetailPage?

File: src/main/java/page/mentors/EditMentorsDetailsPage.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        clickElement(disableButton);
+        return this;
+    }
+    public EditMentorsDetailsPage clearFields(){
+        clickElement(clearButton);
+        return this;
+    }
+    public  EditMentorsDetailsPage saveMentor(){
+        clickElement(saveButton);
+        return this;
```
Comment (OlekLiakh) [inline src/main/java/page/mentors/EditMentorsDetailsPage.java]: After clicking on the button "Save" or "Exclude", a request is sent, and during this time, the buttons "Save" and "Exclude" are disabled. Then if a request is successful - the system redirects the user to the List of Mentors.

So you'll be have another return type

File: src/main/java/page/mentors/EditMentorsDetailsPage.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        clickElement(deleteCourseButton);
+        return this;
+    }
+    public EditMentorsDetailsPage addCourse(){
+        clickElement(addCourseButton);
+        return this;
+    }
+    public EditMentorsDetailsPage removeMentor(){
+        clickElement(disableButton);
+        return this;
```
Comment (OlekLiakh) [inline src/main/java/page/mentors/EditMentorsDetailsPage.java]: After clicking on the button "Save" or "Exclude", a request is sent, and during this time, the buttons "Save" and "Exclude" are disabled. Then if a request is successful - the system redirects the user to the List of Mentors.

So you'll be have another return type

File: src/main/java/page/mentors/EditMentorsDetailsPage.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
+package page.mentors;
+
+import constants.Locators;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import page.base.Page;
+
+public class EditMentorsDetailsPage extends Page implements Locators.EditMentorsDetailsPage {
+    public EditMentorsDetailsPage(WebDriver driver) {
```
Comment (Hetman18) [inline src/main/java/page/mentors/EditMentorsDetailsPage.java:10]: We agreed that we all will import 'Locators class', so you must import class, not implements.
For example:
"import static constants.Locators.Courses.*;"

File: src/main/java/page/mentors/EditMentorsDetailsPage.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        super(driver);
+    }
+    @FindBy (xpath = MENTORS_DETAILS_TAB_XPATH)
+    WebElement mentorsDetailsTab;
+    @FindBy (xpath =EDIT_MENTOR_TAB_XPATH)
+    WebElement editMentorsTab;
+    @FindBy (xpath =ARROW_BUTTON_XPATH)
```
Comment (Hetman18) [inline src/main/java/page/mentors/EditMentorsDetailsPage.java]: Maybe you should make a 'private' variable?

File: src/main/java/page/mentors/EditMentorsDetailsPage.java [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    @FindBy (xpath =SAVE_BUTTON_XPATH)
+    WebElement saveButton;
+
+    public EditMentorsDetailsPage viewMentorsDetailTab(){
+        clickElement(mentorsDetailsTab);
+        return this;
+    }
+    public EditMentorsDetailsPage viewEditMentorsTab(){
+        clickElement(editMentorsTab);
+        return this;
```
Comment (Hetman18) [inline src/main/java/page/mentors/EditMentorsDetailsPage.java]: As I understand, this method should be on 'MentorsDetailsTab' page?

**Review by Hetman18 — APPROVED** [review]

**Review by OlekLiakh — APPROVED** [review]

**Pull request merged**